common/models/user.js

```diff
@@ -580,21 +580,11 @@ module.exports = function(User) {
 
     UserModel.on('attached', function() {
       UserModel.afterRemote('confirm', function(ctx, inst, next) {
-        if (ctx.req) {
-          // replacement for deprecated req.param()
-          var params = ctx.req.params;
-          var body = ctx.req.body;
-          var query = ctx.req.query;
-          var redirectUrl =
-            params && params.redirect !== undefined ? params.redirect :
-            body && body.redirect !== undefined ? body.redirect :
-            query && query.redirect !== undefined ? query.redirect :
-            undefined;
-
-          ctx.res.redirect(redirectUrl);
-        } else {
-          next(new Error('transport unsupported'));
+        if (ctx.args.redirect !== undefined) {
+          ctx.res.location(ctx.args.redirect);
+          ctx.res.status(302);
         }
+        next();
```

Comment:
`ctx.res` may be undefined when non-REST transport like WebSockets is used.

Here is a mockup of what I'd like to see here: 

```js
if (ctx.args.redirect !== undefined) {
  if (!ctx.res) {
    return next(new Error('The transport does not support HTTP redirects.'));
  }
  // redirect via ctx.res.location & status
}
next();
```